runtime/tests/checker/entitlements_test.go

```diff
@@ -5730,3 +5751,240 @@ func TestCheckIdentityMapping(t *testing.T) {
 		require.Equal(t, 0, auth.Entitlements.Len())
 	})
 }
+
+func TestCheckEntitlementErrorReporting(t *testing.T) {
+	t.Run("three or more conjunction", func(t *testing.T) {
+		t.Parallel()
+		checker, err := ParseAndCheckWithOptions(t, `
+		entitlement X
+		entitlement Y
+		entitlement Z
+		entitlement A
+		entitlement B
+		struct S {
+			view access(X, Y, Z) fun foo(): Bool {
+				return true
+			}
+		}
+		fun bar(r: auth(A, B) &S) {
+			r.foo()
+		}
+	`, ParseAndCheckOptions{Config: &sema.Config{SuggestionsEnabled: true}})
+
+		errs := RequireCheckerErrors(t, err, 1)
+		require.IsType(t, &sema.InvalidAccessError{}, errs[0])
+		assert.Equal(
+			t,
+			errs[0].(*sema.InvalidAccessError).RestrictingAccess,
+			sema.NewEntitlementSetAccess(
+				[]*sema.EntitlementType{
+					checker.Elaboration.EntitlementType("S.test.X"),
+					checker.Elaboration.EntitlementType("S.test.Y"),
+					checker.Elaboration.EntitlementType("S.test.Z"),
+				},
+				sema.Conjunction,
+			),
+		)
+		assert.Equal(
+			t,
+			errs[0].(*sema.InvalidAccessError).PossessedAccess,
+			sema.NewEntitlementSetAccess(
+				[]*sema.EntitlementType{
+					checker.Elaboration.EntitlementType("S.test.A"),
+					checker.Elaboration.EntitlementType("S.test.B"),
+				},
+				sema.Conjunction,
+			),
+		)
+		assert.Equal(
+			t,
+			errs[0].(*sema.InvalidAccessError).SecondaryError(),
+			"reference needs all of entitlements `X`, `Y`, and `Z`",
+		)
+	})
+
+	t.Run("has one entitlement of three", func(t *testing.T) {
+		t.Parallel()
+		checker, err := ParseAndCheckWithOptions(t, `
+		entitlement X
+		entitlement Y
+		entitlement Z
+		entitlement A
+		entitlement B
+		struct S {
+			view access(X, Y, Z) fun foo(): Bool {
+				return true
+			}
+		}
+		fun bar(r: auth(A, B, Y) &S) {
+			r.foo()
+		}
+	`, ParseAndCheckOptions{Config: &sema.Config{SuggestionsEnabled: true}})
```

Comment:
Please chop down long argument lists, i.e. add line breaks between arguments, and split complex/long composite literals onto separate lines, too. That makes it easier to read and refactor the code